File: 프로그래머스/0/181890. 왼쪽 오른쪽/왼쪽 오른쪽.cs
using System;
using System.Collections.Generic;

public class Solution 
{
    public string[] solution(string[] str_list) 
    {
        int count = 0; 
        List<string> strList = new List<string>(str_list); // 먼저 리스트에 str_list배열을 옮겨담는다.
        List<string> answerList = new List<string>(); // 반환할 값을 담을 리스트를 만든다
        
        for (int i=0; i<str_list.Length; i++) { // i을 0부터 str의 길이만큼 반복하면서
            if(str_list[i] == "l") { // str_list의 i번째 인덱스의 값이 l이라면
                answerList = strList.GetRange(0, i); // answerList에 strList의 0부터 count 값을 떼와서 붙히기
                break; // 더 순환할 필요가없으므로 for문 탈출
            }
            else if(str_list[i] == "r") { // str_list의 i번째 인덱스의 값이 r이라면
                // answerList에 strList의 count+1부터 strList.Count - count -1 만큼의 값을 떼와서 붙히기
                answerList = strList.GetRange(i+1,strList.Count - i -1);
                break; // 더 순환할 필요가없으므로 for문 탈출
            }
        }
        string[] answer = answerList.ToArray();
        return answer;
    }
}

File: 프로그래머스/0/181890. 왼쪽 오른쪽/왼쪽 오른쪽.cpp
#include <string>
#include <vector>
#include <iostream>
#include <algorithm>

using namespace std;

vector<string> solution(vector<string> str_list) 
{
    // "l" 또는 "r"가 처음 나타나는 위치
    auto it = find_if(str_list.begin(), str_list.end(),
                      [](const string& s){ return s == "l" || s == "r"; });

    if (it == str_list.end()) return {};            // l/r 없으면 빈 벡터
    if (*it == "l") return {str_list.begin(), it};  // l이면 왼쪽만
    return {next(it), str_list.end()};              // r이면 오른쪽만
}